spec.bs

```diff
@@ -938,10 +938,11 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
         [=serializing a JavaScript value to a JSON string=], given |result|.
     * To handle an error, set |auctionConfig|'s [=auction config/seller signals=] to failure.
 1. If |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"] [=map/exists=]:
-  1. [=Handle an input promise in configuration=] given |auctionConfig| and |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"]:
+  1. Set |auctionConfig|'s [=auction config/expects direct from seller signals header ad slot=] to true.
+  1. [=Handle an input promise in configuration=] given |auctionConfig| and
+    |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"]:
     * To parse the value |result|:
       1. Set |auctionConfig|'s [=auction config/direct from seller signals header ad slot=] to |result|.
-      1. Set |auctionConfig|'s [=auction config/expects direct from seller signals header ad slot=] to true.

```

Comment:
It seems our implementation uses [=auction config/expects direct from seller signals header ad slot=] to wait for fetch()'s result. Is that correct? I think I misunderstood what the boolean is used for.

Reply:
The directFromSellerSignalsHeaderAdSlot implementation doesn't wait for fetch() in any way -- it only waits for the directFromSellerSignalsHeaderAdSlot promise to resolve. The ad tech JavaScript code calling runAdAuction() with directFromSellerSignalsHeaderAdSlot is responsible for not resolving the directFromSellerSignalsHeaderAdSlot promise until the fetch() promise has been resolved, or alternatively, not calling runAdAuction() until the fetch() promise has resolved.

[=auction config/expects direct from seller signals header ad slot=] set to true means there is an active, unresolved directFromSellerSignalsHeaderAdSlot promise that is blocking the auction from completing. directFromSellerSignalsHeaderAdSlot set to false means that either the directFromSellerSignalsHeaderAdSlot promise has resolved, or directFromSellerSignalsHeaderAdSlot wasn't passed to the runAdAuction() call.

Reply:
Thanks for the explaination! It's finally clear to me. I thought our runAdAuction implementation was responsible for waiting for the fetch result in some way, separate from waiting for directFromSellerSignalsHeaderAdSlot promise.
Now in the spec, I think this [=auction config/expects direct from seller signals header ad slot=] is no longer necessary, since we can have the method [wait until configuration input promises resolve ](https://wicg.github.io/turtledove/#wait-until-configuration-input-promises-resolve) that waits for promises.

Reply:
What happens if `directFromSellerSignalsHeaderAdSlot` resolves before the `fetch()` response resolves? Can we mention that in the note that @qingxinwu added in this PR? Is _that_ the "race condition" that we talked about?

Reply:
yes, that's the race condition.